packages/docusaurus-utils-validation/src/validationSchemas.ts

```diff
@@ -14,10 +14,9 @@ export const PluginIdSchema = Joi.string()
 const MarkdownPluginsSchema = Joi.array()
   .items(
     Joi.array()
-      // TODO, this allows [config,fn] too?
-      .items(Joi.function().required(), Joi.object().required())
-      .length(2),
+      .ordered(Joi.function().required(), Joi.object().required()),
     Joi.function(),
+    Joi.object(),
   )
   .default([]);
```

Comment:
This completes the TODO for validating 'babel style' input and fixes the validation issue

Reply:
ordered is really cool

Reply:
yes, looked for that but couldn't find it. Joi doc is a pain :D